### ISSUE TITLE
Giving a name to a TileSetSource in the editor actually modifies the resource_name property

### ISSUE THREAD

**Issue by mrcdk** [godotengine/godot#73970]
### Godot version

v4.0.rc5.official [6296b4600]

### System information

Manjaro Linux

### Issue description

In the editor when giving a `name` to a `TileSetSource` actually changes the `Resource.resource_name` internally and, if I didn't check the `TileSet`'s `.tres` file, I wouldn't find out that as there's no API to do that otherwise. 

I'm referring to this property:

![image](https://user-images.githubusercontent.com/483531/221406941-b50c142e-4032-49f6-becd-53c941f3a64a.png)

I see two options:

- Change the `name` property in the editor to `resource_name` to make it clear what's being modified, or
- Expose to the `TileMap`/`TileSetSource` API a way to access and modify it at runtime.

### Steps to reproduce

N/A

### Minimal reproduction project

N/A

**Comment by KoBeWi** [comment]
But you can access and modify it at runtime...

**Comment by mrcdk** [comment]
> But you can access and modify it at runtime...

Yes, my bad, I didn't explain myself correctly. I meant adding a `TileSetSource.name` property or `TileSetSource.get_name(id)`/`TileSetSource.set_name(id)` getter and setter like what's already available to get/set a terrain name [`TileSet.get_terrain_name()`](https://docs.godotengine.org/en/latest/classes/class_tileset.html#class-tileset-method-get-terrain-name) [`TileSet.set_terrain_name()`](https://docs.godotengine.org/en/latest/classes/class_tileset.html#class-tileset-method-set-terrain-name)

I'll update the original post with this :+1:
